versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksBaseValue.java

```diff
@@ -0,0 +1,225 @@
+  public C dt(long dt) {
+    this.datetime = dt;
+    return (C) this;
+  }
+
+  Id getId() {
+    return id;
+  }
+
+  /**
+   * Converts a Stream of Ids into a List Entity.
+   * @param idStream the stream of Id to convert.
+   * @return the List Entity.
+   */
+  Entity toEntity(Stream<Id> idStream) {
+    return Entity.ofList(idStream.map(Id::toEntity).collect(Collectors.toList()));
+  }
+
+  /**
+   * Retrieves an Id at 'position' in a Stream of Ids as an Entity.
+   * @param idStream the stream of Id to convert.
+   * @param position the element in the Stream to retrieve.
+   * @return the List Entity.
+   */
+  Entity toEntity(Stream<Id> idStream, int position) {
+    return idStream.skip(position).findFirst().orElseThrow(NoSuchElementException::new).toEntity();
+  }
+
+  /**
+   * Evaluates if the stream of Id meets the Condition Function.
+   * @param function the function of the ConditionExpression to evaluate.
+   * @param stream The Ids to evaluate against the function
+   * @return true if the condition is met
+   */
+  boolean evaluateStream(Function function, Stream<Id> stream) {
+    // EQUALS will either compare a specified position or the whole stream as a List.
+    if (function.isEquals()) {
+      final ExpressionPath.PathSegment pathSegment = function.getPath().getRoot().getChild().orElse(null);
+      if (pathSegment == null) {
+        return toEntity(stream).equals(function.getValue());
+      } else { // compare complete list
+        if (pathSegment.isPosition()) { // compare individual element of list
+          int position = pathSegment.asPosition().getPosition();
+          return toEntity(stream, position).equals(function.getValue());
+        }
+      }
+    } else if (function.isSize()) {
+      return (stream.count() == function.getValue().getNumber());
+    }
+
+    return false;
+  }
+
+  /**
+   * A utility to aid evaluation of the Function in checking for equality on
+   * a leaf {@link com.dremio.nessie.versioned.impl.condition.ExpressionPath.NameSegment}.
+   * @param nameSegment the NameSegment that is checked to be childless
+   * @param function the condition function to check it is for equality
+   * @return true if both nameSegment is childless and function has an equality operator
+   */
+  static boolean nameSegmentChildlessAndEquals(ExpressionPath.NameSegment nameSegment, Function function) {
+    return !nameSegment.getChild().isPresent()
+        && function.isEquals();
+  }
+
+  @Override
+  public boolean evaluate(Condition condition) {
+    for (Function function: condition.getFunctions()) {
+      if (function.getPath().getRoot().isName()) {
+        final ExpressionPath.NameSegment nameSegment = function.getPath().getRoot().asName();
+        if (!evaluateSegment(nameSegment, function)) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public abstract boolean evaluateSegment(ExpressionPath.NameSegment nameSegment, Function function);
```

Comment:
As mentioned earlier, this shouldn't be on the interface as it's only called from evaluate() - why would it need to be on the interface?

Reply:
Agreed. Updated as suggested and renamed it to evaluateFunction. Also reduced dependencies by removing NameSegment from parameters. It comes from Function anyway, so no need to pass separately.